Mathlib/Combinatorics/SimpleGraph/Coloring.lean

```diff
@@ -248,46 +256,72 @@ theorem chromaticNumber_bddBelow : BddBelow { n : ℕ | G.Colorable n } :=
 #align simple_graph.chromatic_number_bdd_below SimpleGraph.chromaticNumber_bddBelow
 
 theorem chromaticNumber_le_of_colorable {n : ℕ} (hc : G.Colorable n) : G.chromaticNumber ≤ n := by
-  rw [chromaticNumber]
+  rw [hc.chromaticNumber_def]
+  norm_cast
   apply csInf_le chromaticNumber_bddBelow
   constructor
   exact Classical.choice hc
 #align simple_graph.chromatic_number_le_of_colorable SimpleGraph.chromaticNumber_le_of_colorable
 
+theorem colorable_of_chromaticNumber_le {n : ℕ} (h : G.chromaticNumber ≤ n) : G.Colorable n := by
+  rw [chromaticNumber] at h
+  have h : ⨅ m : {m | G.Colorable m}, ↑m ≤ (n : ℕ∞) := by
+    rw [iInf_subtype]
+    exact h
+  have : ⨅ m : {m | G.Colorable m}, ↑m < (⊤ : ℕ∞) := by
+    exact Trans.trans h (show n < (⊤ : ℕ∞) by rw [← cmp_eq_gt_iff]; rfl)
+  rw [ENat.iInf_coe_lt_top] at this
+  have h : sInf {m | G.Colorable m} ≤ n := by
+    rw [← ENat.coe_iInf, ← sInf_range] at h
+    simp only [Set.coe_setOf, Set.mem_setOf_eq, Subtype.range_coe_subtype, Nat.cast_le] at h
+    exact h
+  have := Nat.sInf_mem (Set.nonempty_coe_sort.mp this)
+  rw [Set.mem_setOf_eq] at this
+  exact this.mono h
+
+theorem chromaticNumber_le_iff_colorable {n : ℕ} : G.chromaticNumber ≤ n ↔ G.Colorable n :=
+  ⟨colorable_of_chromaticNumber_le, chromaticNumber_le_of_colorable⟩
```

Comment:
Can you merge these two into one? I don't think there's any benefit to having `colorable_of_chromaticNumber_le` as a separate implication.

Reply:
I don't mind them being merged. I also wouldn't mind if someone simplified the proof, since it's a hack job. (In case that expression isn't known outside the US, it means it was done cheaply and quickly at the expense of quality.) I remember it being annoying manipulating this inf expression -- maybe there's missing API.